pkg/reconciler/buildrun/resources/results.go

```diff
@@ -52,6 +54,11 @@ func updateBuildRunStatusWithOutputResult(ctx context.Context, buildRun *build.B
 			} else {
 				buildRun.Status.Output.Size = size
 			}
+		case generateOutputResultName(imageVulnerabilities):
+			if err := json.Unmarshal([]byte(result.Value.StringVal), &buildRun.Status.Output.Vulnerabilities); err != nil {
```

Comment:
This is the other side of the vulnerabilities serialization using an (size-wise) inefficient json-serialization.